dev/devicelab/bin/run.dart

```diff
@@ -80,7 +80,8 @@ final ArgParser _argParser = new ArgParser()
         ' - the path to a Dart file corresponding to a task, which resides in bin/tasks. Example: bin/tasks/complex_layout__start_up.dart.\n'
         '\n'
         'This option may be repeated to specify multiple tasks.',
-    callback: (List<String> value) {
+    callback: (Object parsed) {
+      final List<String> value = parsed;
```

Comment:
I really don't like this one. The problem here is that the `args` package isn't typed. That's fine, authors don't have to use types if they don't want to. But then the lack of typing is infecting this package, which _is_ typed. We know that the value will be `List<String>` because, well, that's what it does (it's unfortunately not documented either, so it's hard to really tell for sure without looking at the code).

I think if the target is entirely untyped, as in this case, it should be fine to specify the type in the argument.

Reply:
> But then the lack of typing is infecting this package, which is typed

Academic work on gradual typing "just makes this work" by automatically wrapping, but there's a fair bit of evidence that the runtime cost is prohibitive in practice.  We've tried to take a practical compromise, but it does make the interface between typed and untyped code more awkward.

I can look into making changes on the ArgParser side.

Maybe the `callback` parameter should just be typed as `Function`?  Really, I want to have a type for "any arity N function", but we don't have that right now.  

Another option would be to make addOption generic.

Reply:
cc @nex3 @kevmoo for comment on the ArgParser.addOption interface?

Reply:
cc @munificent 

The problem isn't that `args` chooses to eschew types (it's as typed as anything), but that the type system isn't powerful enough to express the right type here. The `callback` will be called with a `String` if the `allowMultiple` parameter is `false`, and with a `List<String>` if it's `true`. This wasn't a big deal when the package was written, but it definitely feels weird now. I'm not sure what the best solution is... it seems tricky to solve this well without at least union types or overloads.

Reply:
What about making it generic over the argument type, and letting it get inferred from the provided callback?  The client of the API gets better static checking inside of their callback, and no worse static checking of their use of the API.

Reply:
I think what I'd prefer is if the type we're trying to match against is an "implicit dynamic", then anything matches (with a runtime type check).

Reply:
Yes, I think there's a need for a type of all functions of arity N that doesn't feed into inference: something like `? -> void` meaning a unary function returning void.  The type `Null -> void` is the type of all unary functions returning void, but it's not very friendly for inference.  This probably works fine for flutter code, since you type all of your closure arguments explicitly, but not as well for other coding styles.  `Function` is the closest we have to this now, but it doesn't have a return type, and doesn't specify arity.

Reply:
I'll fix this in argparser.

Reply:
@nex3 has a PR fixing this, I'll roll this in when it gets published.